xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-api/src/main/java/com/xwiki/macros/confluence/internal/ConfluenceSpaceUtils.java

```diff
@@ -77,23 +87,29 @@ public EntityReference getSloppySpace(String spaceKeyOrRef)
                 return confluenceSpaceResolver.getSpace(contextProvider.get().getDoc().getDocumentReference());
             }
 
-            if (spaceKeyOrRef.indexOf(':') != -1 || spaceKeyOrRef.indexOf('.') != -1) {
-                // This is a XWiki reference
-                EntityReference spaceRef = resolver.resolve(spaceKeyOrRef, EntityType.SPACE);
-                EntityReference webHome = new EntityReference("WebHome", EntityType.DOCUMENT, spaceRef);
-                if (contextProvider.get().getWiki().exists(new DocumentReference(webHome), contextProvider.get())) {
-                    // the home page of this space exists
-                    return spaceRef;
-                }
+            if (spaceKeyOrRef.startsWith("confluenceSpace:")) {
+                // The spaceKeyOrRef might be prefixed with confluenceSpace: when at migration time we didn't know
+                // where the space is. In the meantime it might have been migrated, and we should check if the space
+                // is available in XWiki
+                String unprefixedSpaceKey = spaceKeyOrRef.substring(spaceKeyOrRef.indexOf(':') + 1);
+                return confluenceSpaceKeyResolver.getSpaceByKey(unprefixedSpaceKey);
+            }
+            EntityReference spaceRef = resolver.resolve(spaceKeyOrRef, EntityType.SPACE);
+            Query checkSpaceExistence =
+                queryManager.createQuery("SELECT count(reference) FROM XWikiSpace s where s.reference=:reference",
+                    Query.HQL);
+            checkSpaceExistence.bindValue("reference", serializer.serialize(spaceRef));
+            long count = (long) checkSpaceExistence.execute().get(0);
+            if (count > 0) {
+                return spaceRef;
             }
 
             return confluenceSpaceKeyResolver.getSpaceByKey(spaceKeyOrRef);
         } catch (ConfluenceResolverException e) {
             logger.warn("Could not convert space [{}] to an entity reference", spaceKeyOrRef, e);
-        } catch (XWikiException e) {
-            logger.warn("Could not check document [{}] existence", spaceKeyOrRef, e);
+        } catch (QueryException e) {
+            logger.warn("Query failed", e);
```

Comment:
Could have a more descriptive message... You are also passing a throwable to a warning level log. If i'm not mistaken, passing the error as the last parameter only works for `error` level logs.